oat-sa/extension-tao-itemqti-pci
# Feature/TR-4575/Render maths expression in prompt using MathQuill

### Related to: https://oat-sa.atlassian.net/browse/TR-4575

### Summary
Bring support to Math Entry PCI for rendering Math expressions in the prompt.

### Details
MathJax cannot be bundled with a PCI as it is not properly scoped. For this reason, we need to use another renderer for having Maths expressions in the prompt of the PCI.

This PCI is already including such a rendered. Indeed, the Maths expressions are managed by MathQuill, which supports LaTeX.

For rendering Maths in prompt, we can use it as well. However, need to make sure the expressions are expressed in LaTeX only. A 3rd-party library is needed to convert MathML. This is done thanks to [`mathml-to-latex`](https://github.com/qtikit/mathml-to-latex).

### How to test
- create/update an item and a Math Entry PCI
- add a Math expression in the prompt (better to add both LaTeX and MathML expressions for better testing)
- save, and edit again: the Math expression should be rendered properly
- create/update a test with the ite,
- publish the test and take the test: the Math expression should render fine


### Technical details
- Even if MathJax is not used by the feature, you still have to install it. Otherwise, the authoring won't propose the Math expression container.
- To add a Math expression to the prompt, you need to select the tool in the editor toolbar, then you can insert the Math expression in the input field from the right side panel.
<img width="1434" alt="image" src="https://user-images.githubusercontent.com/1500098/190375797-0cdbba8e-6fce-48ab-8f3f-7c5048fb0255.png">
